src/controllers/CategoryController.ts

```diff
@@ -9,22 +9,21 @@ export class CategoryController {
     @inject(CategoryService) private categoryService: CategoryService
   ) {}
 
-  public async getAllCategories(
-    req: Request,
-    res: Response
-  ): Promise<Category[]> {
+  public async getAllCategories(req: Request, res: Response) {
     try {
       const Categories = await this.categoryService.getAllCategories()
-
-      res.json(Categories)
-      return Categories
+      if (!Categories) {
+        return res.status(404).json({ error: 'could not find categories' })
+      }
+      return res.status(200).json(Categories)
     } catch (error: any) {
-      res.status(500).json({ error: error.message })
-      throw new Error(`Error retrieving categories`)
+      res
+        .status(500)
+        .json({ error: 'Internal server error, please try again later' })
```

Comment:
Too generic statement - bad for debugging, you need to be more precise explaining where the error is.

Reply:
I will fix them